web/text_layer_builder.js

```diff
@@ -101,10 +101,10 @@ class TextLayerBuilder {
           mustRescale,
           mustRotate,
         });
-        this.show();
         this.#scale = scale;
         this.#rotation = rotation;
       }
+      this.show();
```

Comment:
This is a really good find, since it actually fixes an existing bug that I've seen intermittently!
(Where the textLayer sometimes would never become visible again after zooming, if you changed zoom-level multiple times for a slowly rendering document).

Reply:
Yep, my way to test zoom+text_layer is to ctrl+a and zoom to see if everything is ok.